### Day 1 SQL Homework


![](https://miro.medium.com/v2/resize:fit:749/1*DN0iewN5WFWgrXs5s5cLjQ.jpeg)

In [5]:
import pandas as pd

In [20]:
titanic = pd.read_csv(r'https://raw.githubusercontent.com/lucchesia7/coding-temple-da-sql-day-1/main/day_1/data/titanic.csv')
titanic.head()

Survived  Pclass                                               Name  \
0         0       3                             Mr. Owen Harris Braund   
1         1       1  Mrs. John Bradley (Florence Briggs Thayer) Cum...   
2         1       3                              Miss. Laina Heikkinen   
3         1       1        Mrs. Jacques Heath (Lily May Peel) Futrelle   
4         0       3                            Mr. William Henry Allen   

      Sex   Age  Siblings/Spouses_Aboard  Parents/Children_Aboard     Fare  
0    male  22.0                        1                        0   7.2500  
1  female  38.0                        1                        0  71.2833  
2  female  26.0                        0                        0   7.9250  
3  female  35.0                        1                        0  53.1000  
4    male  35.0                        0                        0   8.0500

In [21]:
titanic.columns = titanic.columns.str.lower().str.replace(' ', '_').str.replace('/', '_')
titanic.head()

survived  pclass                                               name  \
0         0       3                             Mr. Owen Harris Braund   
1         1       1  Mrs. John Bradley (Florence Briggs Thayer) Cum...   
2         1       3                              Miss. Laina Heikkinen   
3         1       1        Mrs. Jacques Heath (Lily May Peel) Futrelle   
4         0       3                            Mr. William Henry Allen   

      sex   age  siblings_spouses_aboard  parents_children_aboard     fare  
0    male  22.0                        1                        0   7.2500  
1  female  38.0                        1                        0  71.2833  
2  female  26.0                        0                        0   7.9250  
3  female  35.0                        1                        0  53.1000  
4    male  35.0                        0                        0   8.0500

In [22]:
connection = 'postgresql://dbzscqgx:JSiCY0dwvCjusi0qV2tzZM7PexVv_O2o@peanut.db.elephantsql.com/dbzscqgx'

In [23]:
titanic.to_sql('titanic', con = connection, if_exists = 'fail')

887

### **Question 1**: Count how many rows you have.

```sql

SELECT COUNT(*) FROM titanic

```

Answer = 887



### **Question 2**: How many people survived?

```sql

SELECT SUM(survived) FROM titanic

```

Answer = 342

### **Question 3**: What passenger class has the largest population?

```sql

SELECT pclass, COUNT(pclass) FROM titanic
GROUP BY pclass
ORDER BY COUNT(pclass)
DESC
LIMIT 1;

```

Answer = Pclass 3 with 487 Passengers